server/datastore/mysql/hosts.go

```diff
@@ -314,6 +314,11 @@ func (ds *Datastore) DeleteHost(ctx context.Context, hid uint) error {
 			return ctxerr.Wrapf(ctx, err, "deleting pack_targets for host %d", hid)
 		}
 
+		_, err = tx.ExecContext(ctx, `DELETE FROM windows_updates WHERE host_id = ?`, hid)
```

Comment:
Since most of the time the use of FKs is discouraged because of performance reasons, we might want to start thinking of a way to structure clean up operations like this so that they can be performed asynchronously. For example, maybe keep a table with removed hosts and then have a cron job that uses that table to remove all related data.